### PR TITLE
StringExtensions tests (Droid)

### PR DESCRIPTION
**Pull request by SavikPavel**
## Description

- StringExtensions reworked (Droid)
- StringExtensions tests added (Droid)
- BoolToViewStateConverter removed. Functionality moved to GoneConverter

### API Changes

Changed:
 - object BoolToViewStateConverter.GoneConverter => GoneConverter.Instance

### Platforms Affected

- Android

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.Common.Droid/Softeq.XToolkit.Common.Droid.csproj [hunk 1]
```diff
@@ -24,6 +24,7 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <LangVersion>latest</LangVersion>
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply Softeq.XToolkit.Common.Droid/Softeq.XToolkit.Common.Droid.csproj]: Removed

File: Softeq.XToolkit.Common.Droid.Tests/Softeq.XToolkit.Common.Droid.Tests.csproj [hunk 2]
```diff
@@ -50,6 +51,7 @@
     <AndroidUseSharedRuntime>False</AndroidUseSharedRuntime>
     <AndroidLinkMode>SdkOnly</AndroidLinkMode>
     <EmbedAssembliesIntoApk>True</EmbedAssembliesIntoApk>
+    <LangVersion>latest</LangVersion>
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply Softeq.XToolkit.Common.Droid.Tests/Softeq.XToolkit.Common.Droid.Tests.csproj]: Removed

File: Softeq.XToolkit.Common.Droid/Converters/GoneConverter.cs [hunk 3]
```diff
@@ -11,6 +11,8 @@ namespace Softeq.XToolkit.Common.Droid.Converters
     /// </summary>
     public class GoneConverter : IConverter<ViewStates, bool>
     {
+        public static GoneConverter Instance { get; } = new GoneConverter();
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply Softeq.XToolkit.Common.Droid/Converters/GoneConverter.cs]: Fixed

File: Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs [hunk 4]
```diff
@@ -12,48 +13,80 @@ namespace Softeq.XToolkit.Common.Droid.Extensions
...
+        /// /// <exception cref="T:System.ArgumentOutOfRangeException">
+        ///     <paramref name="startingIndex"/> parameter must be less than the size of the string and greater or equal than 0.
+        ///     - and -
+        ///     <paramref name="length"/> parameter must be less than the size of the string and greater or equal than 0.
+        ///     - and -
+        ///     sum of these params must be less than the size of the string.
+        /// </exception>
+        public static SpannableString FormatSpannable(
+            this string unformattedString,
+            int startingIndex,
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs:35]: Sounds like a great idea to me!

[the remark this answers is not in the document]
Comment (SavikPavel) [reply Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs]: Reworked

File: Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs [hunk 5]
```diff
@@ -12,48 +14,86 @@ namespace Softeq.XToolkit.Common.Droid.Extensions
...
+        ///     End index of the <paramref name="textRange"/> must be less than the size of the string.
+        /// </exception>
+        public static SpannableString FormatSpannable(
+            this string unformattedString,
+            TextRange textRange,
+            params Object[] spans)
         {
+            if (unformattedString == null)
+            {
+                throw new ArgumentNullException($"{nameof(unformattedString)} cannot be null");
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs:43]: can be simplified as `new ArgumentNullException(nameof(unformattedString))` as a result:

```
System.ArgumentNullException: Value cannot be null.
Parameter name: unformattedString
...
```

**Review by wcoder — APPROVED** [review]
Only one comment:

File: Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs (hunk 5 above)
Comment (SavikPavel) [reply Softeq.XToolkit.Common.Droid/Extensions/StringExtensions.cs]: Reworked

**Review by wcoder — APPROVED** [review]

**Pull request merged**